app/javascript/components/public_health/AdvancedFilter.js

```diff
@@ -0,0 +1,608 @@
+      .catch(() => {
+        toast.error('Failed to save filter.');
+      })
+      .then(response => {
+        toast.success('Filter successfully saved.');
+        this.setState({ activeFilter: response.data, savedFilters: [...self.state.savedFilters, response.data] });
+      });
+  }
+
+  // Update an existing filter
+  update() {
+    let self = this;
+    axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+    axios
+      .put(window.BASE_PATH + '/user_filters/' + this.state.activeFilter.id, { active: this.state.active })
+      .catch(() => {
+        toast.error('Failed to update filter.');
+      })
+      .then(response => {
+        toast.success('Filter successfully updated.');
+        this.setState({
+          activeFilter: response.data,
+          savedFilters: [
+            ...self.state.savedFilters.filter(filter => {
+              return filter.id != response.data.id;
+            }),
+            response.data,
+          ],
+        });
+      });
+  }
+
+  // Delete an existing filter
+  delete() {
+    let self = this;
+    const id = this.state.activeFilter.id;
+    axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+    axios
+      .delete(window.BASE_PATH + '/user_filters/' + id)
+      .catch(() => {
+        toast.error('Failed to delete filter.');
+      })
+      .then(() => {
+        toast.success('Filter successfully deleted.');
+        this.setState({
+          show: false,
+          applied: false,
+          activeFilter: null,
+          savedFilters: [
+            ...self.state.savedFilters.filter(filter => {
+              return filter.id != id;
+            }),
+          ],
+        });
+      });
+  }
+
+  // Render the options for the select that represents fields to filter on
+  renderOptions(current, index) {
+    return (
+      <Form.Group key={index + 'opkeygroup'} className="py-0 my-0">
+        <Form.Control
+          as="select"
+          value={current}
+          className="py-0 my-0"
+          size="sm"
+          onChange={event => {
+            this.changeFilterOption(index, event.target.value);
+          }}>
+          {this.state.filterOptions.map((option, op_index) => {
+            return (
+              <option key={index + 'opkeyop' + op_index} value={option.name} disabled={option.type === 'blank'}>
+                {option.title}
+              </option>
+            );
+          })}
+        </Form.Control>
+      </Form.Group>
+    );
+  }
+
+  // Render date specific options
+  renderDateOptions(current, index) {
+    return (
+      <Form.Group key={index + 'opkeygroup'} className="py-0 my-0">
+        <Form.Control
+          as="select"
+          value={current}
+          className="py-0 my-0"
+          size="sm"
+          onChange={event => {
+            this.changeFilterDateOption(index, event.target.value);
+          }}>
+          <option value="within">within</option>
```

Comment:
was thinking of maybe using something along the lines of "between MM/DD/YYYY and MM/DD/YYYY" instead of "within MM/DD/YYYY to MM/DD/YYYY" for dates?

Reply:
Yeah, that **exact** thought occurred to me last night. What do others think, @kylietmo ?

Reply:
The issue is we making sure it's clear if the dates are inclusive or not. When you say between some people may interpret that differently. I'd say something like "from MM/DD/YYYY to MM/DD/YYYY".

Reply:
I think I'm going to defer to the mockups on this one, and not make this change. I think `within` does make sense.